### PR TITLE
Add documentation for using private image registries and a default workspace image for Self hosted

### PR DESCRIPTION
**Pull request by lucasvaltl**
## Description
- Explains how to use credentials for private registries and their impact on the privatebaseimageregistryallowlist
- Documents how to set default workspace images

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/pull/12174

## How to test
- read thorugh the added docs - does this make sense? Are the config patches valid????
- 
## Release Notes

```release-note
NONE
```

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2640"></a>

### CODE REVIEW

File: gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+title: Private Registries
+---
+
+<script context="module">
+  export const prerender = true;
+</script>
+
+# Setting an installation-wide default workspace image
+
+This guide describes how to set a default workspace image that will apply to your entire Gitpod Self-Hosted installation. Please refer to the [docker image documentation](../../../config-docker) for more information about workspace images.
```
Comment (MrSimonEmms) [inline gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md:13]:
```suggestion
This guide describes how to set a default workspace image that will apply to your entire Gitpod Self-Hosted installation. Please refer to the [Docker image documentation](../../../config-docker) for more information about workspace images.
```

File: gitpod/docs/self-hosted/latest/advanced/private-registries.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+title: Private Registries
+---
+
+<script context="module">
+  export const prerender = true;
+</script>
+
+# Using private image registries with Gitpod Self-Hosted
+
+This guide outlines the considerations to take when using images for workspaces that originate from private image registries. Please refer to the [docker image documentation](../../../config-docker) for more information about workspace images.
```
Comment (MrSimonEmms) [inline gitpod/docs/self-hosted/latest/advanced/private-registries.md:13]:
```suggestion
This guide outlines the considerations to take when using images for workspaces that originate from private image registries. Please refer to the [Docker image documentation](../../../config-docker) for more information about workspace images.
```

File: gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+> **Note:** there are two ways of setting an image for a workspace: via the installation-wide default `workspaceImage` (mentioned here-) and the [`.gitpod.yml` file](../../../config-gitpod-file). The latter overwrites the former.
```
Comment (MrSimonEmms) [inline gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md:15]:
```suggestion
> **Note:** there are two ways of setting an image for a workspace: via the installation-wide default `workspaceImage` (mentioned here) and the [`.gitpod.yml` file](../../../config-gitpod-file). The `.gitpod.yml` file will take precedence if both are set.
```

Fixes typo. Also, "the latter overwrites the former" might not be clear enough - hopefully, my suggestion is a good starting point

File: gitpod/docs/self-hosted/latest/advanced/private-registries.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+> Note: there are two ways of setting an image for a workspace: via the installation-wide [default `workspaceImage`](./default-workspace-image) and the [`.gitpod.yml` file](../../../config-gitpod-file). The latter overwrites the former.
```
Comment (MrSimonEmms) [inline gitpod/docs/self-hosted/latest/advanced/private-registries.md:15]:
```suggestion
> Note: there are two ways of setting an image for a workspace: via the installation-wide [default `workspaceImage`](./default-workspace-image) and the [`.gitpod.yml` file](../../../config-gitpod-file). The `.gitpod.yml` file will take precedence if both are set.
```

As above

File: gitpod/docs/self-hosted/latest/advanced/private-registries.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+To access private registries, Gitpod needs to have the corresponding access credentials. These can be supplied in the KOTS installation UI:
+
+![registry credentials UI](../../static/images/docs/self-hosted/private-registry-credentials.png)
+
+## Allowlist
+
+Gitpod also has an allow list for private base image registries. Any private image registry that is used throughout an installation needs to be on this allowlist.
+
+> **Important:** Any registry for which credentials are supplied is automatically added to this allowlist!
```
Comment (Pothulapati) [inline gitpod/docs/self-hosted/latest/advanced/private-registries.md:27]: 💯 Stands out like we want.

**Review by MrSimonEmms — COMMENTED** [review]
Good work. I've put a couple of suggestions - most stylistic/typos and there's one suggestion about improving the clarity of the wording (I'm happy to approve without this change).

File: gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+section: self-hosted/latest
+subsection: installation-guides
+title: Private Registries
+---
+
+<script context="module">
+  export const prerender = true;
+</script>
+
+# Setting an installation-wide default workspace image
```
Comment (Pothulapati) [inline gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md:11]: So, If they are using this, there is a good chance that they are also using private registries right for the workspace image? Should we link to the other doc from here? 🤔

Comment (lucasvaltl) [reply gitpod/docs/self-hosted/latest/advanced/default-workspace-image.md:11]: Fixed in the next commit :)

**Review by MrSimonEmms — APPROVED** [review]

**Pull request merged**